Sources/CacheAPI-smf.php

```diff
@@ -49,67 +49,107 @@ public function isSupported($test = false)
 		return parent::isSupported() && $supported;
 	}
 
+	private function readFile($file)
+	{
+		if (($fp = fopen($file, 'rb')) !== false)
+		{
+			if (!flock($fp, LOCK_SH | LOCK_NB))
+			{
+				fclose($fp);
+				return false;
+			}
+			$string = '';
+			while (!feof($fp))
+				$string .= fread($fp, 8192);
+
+			flock($fp, LOCK_UN);
+			fclose($fp);
+
+			return $string;
+		}
+
+		return false;
+	}
+
+	private function writeFile($file, $string)
+	{
+		if (($fp = fopen($file, 'cb')) !== false)
+		{
+			if (!flock($fp, LOCK_EX | LOCK_NB))
+			{
+				fclose($fp);
+				return false;
+			}
+			ftruncate($fp, 0);
+			$bytes = 0;
+			$pieces = str_split($string, 8192);
+			foreach ($pieces as $piece)
+			{
+				if (($val = fwrite($fp, $piece, 8192)) !== false)
+					$bytes += $val;
+				else
+					return false;
+			}
+			fflush($fp);
+			flock($fp, LOCK_UN);
+			fclose($fp);
+
+			return $bytes;
+		}
+
+		return false;
+	}
+
 	/**
 	 * {@inheritDoc}
 	 */
 	public function getData($key, $ttl = null)
 	{
-		$key = $this->prefix . strtr($key, ':/', '-_');
-		$cachedir = $this->cachedir;
+		$file = sprintf('%s/data_%s.cache',
+			$this->cachedir,
+			$this->prefix . strtr($key, ':/', '-_')
+		);
 
 		// SMF Data returns $value and $expired.  $expired has a unix timestamp of when this expires.
-		if (file_exists($cachedir . '/data_' . $key . '.php') && filesize($cachedir . '/data_' . $key . '.php') > 10)
+		if (file_exists($file) && ($raw = $this->readFile($file)) !== false)
 		{
-			// Work around Zend's opcode caching (PHP 5.5+), they would cache older files for a couple of seconds
-			// causing newer files to take effect a while later.
-			if (function_exists('opcache_invalidate'))
-				opcache_invalidate($cachedir . '/data_' . $key . '.php', true);
-
-			if (function_exists('apc_delete_file'))
-				@apc_delete_file($cachedir . '/data_' . $key . '.php');
-
```

Comment:
I think we still need these.  On systems where this would take effect, you have made it so it would keep rebuilding the cache file until it catches.  So in effect it could run the code it should be skipping dozens of times before the opcache/apc clears out.  To help simplify this, we could drop it into a sub method.

I understand we don't do it when clearing the entire cache folder.  Something should be done for that as well.

Reply:
These calls are no longer needed because these files are no longer compiled and cached as bytecode

Reply:
Your right.  I was thinking it still gets cached, but the only one would be the file system caching it.  We still may want to call clearstatcache() then in order to inform php to no longer cache it on that side.